test/template-tests/testcases/CreateUIDefinition/Allowed-Values-Should-Actually-Be-Allowed.test.ps1

```diff
@@ -0,0 +1,36 @@
+﻿param(
+[Parameter(Mandatory=$true)]
+[PSObject]
+$CreateUIDefinitionObject,
+
+[Parameter(Mandatory=$true)]
+[Collections.IDictionary]
+$MainTemplateParameters
+)
+
+$allowedValues = $CreateUIDefinitionObject | 
+    Find-AzureRMTemplate -Key allowedValues -Value * -Like
+
+foreach ($av in $allowedValues) {
+    $parent = $av.ParentObject[0]
+
+    $MainTemplateParam = $MainTemplateParameters[$parent.Name]
+
+    if (-not $MainTemplateParam) {
+        Write-Error "CreateUIDefinition has parameter $($parent.Name), but it is missing from main template parameters "-TargetObject $parent
+        continue
+    }
+
+    $reallyAllowedValues = @(foreach ($v in $av.allowedValues) {
+        if ($v.value) {
+            $v.value
+        } else {
+            $v
+        }
+    })
+
+    if ($MainTemplateParam.defaultValue -and 
+        $reallyAllowedValues -notcontains $MainTemplateParam.defaultValue) {
+        Write-Error "CreateUIDefinition paremter $($parent.Name) does not allow for the default value $($MainTemplateParam.defaultValue) used in the main template"
+   }
+}

```

Comment:
Here I think we need to think "backwards" from mainTemplate... flow looks like:
- for each param in mainTemplate that has allowedValues (grab them)
- find control in CUID that populates that param, make sure allowedValues on that control (if any) are less permissive than in mainTemplate (template could allow 1,2,3 - CUID could allow 1,2, but not 1,2,4)
- CUID could be hidden behind basics or steps function (see below comment)
- CUID could also use an expression like coalesce - in this case we'd need to find all controls in the coalesce stmt and union those for comparison.